app/controllers/utils/ControllerUtils.scala

```diff
@@ -104,26 +144,26 @@ class ControllerUtils @Inject() (configuration: Configuration)(implicit val exec
     (paye_scheme_type, employer_number)
   }
 
-  def retrieveCrendtialsFromNPS(
+  def retrieveCredentialsFromNPS(
     tierConnector: HmrcTierConnectorWrapped,
     year: Int,
     employer_code: String,
     paye_scheme_type: Int
-  )(implicit hc: HeaderCarrier, formats: json.Format[PbikCredentials]): Future[PbikCredentials] =
-    tierConnector.retrieveDataGet(s"$baseURL/$year/$employer_code/$paye_scheme_type")(hc) map { result: HttpResponse =>
-      result.json.validate[PbikCredentials].asOpt.get
+  )(implicit hc: HeaderCarrier): Future[PbikCredentials] =
+    tierConnector.retrieveDataGet(s"${pbikConfig.baseURL}/$year/$employer_code/$paye_scheme_type")(hc) map {
+      result: HttpResponse =>
+        result.json.validate[PbikCredentials].asOpt.get
     }
 
-  def getNPSMutatorSessionHeader(implicit request: Request[AnyContent]): Map[String, String] =
+  def getNPSMutatorSessionHeader(implicit request: Request[_]): Map[String, String] =
     if (request.headers.get(HeaderTags.ETAG).isDefined) {
-      Map(
-        (HeaderTags.ETAG, request.headers.get(HeaderTags.ETAG).getOrElse("0")),
-        (HeaderTags.X_TXID, request.headers.get(HeaderTags.X_TXID).getOrElse("1"))
-      )
+      val etag = request.headers.get(HeaderTags.ETAG).getOrElse(HeaderTags.ETAG_DEFAULT_VALUE)
+      val txid = request.headers.get(HeaderTags.X_TXID).getOrElse(HeaderTags.X_TXID_DEFAULT_VALUE)
```

Comment:
coverage report is showing the getOrElse is not covered by tests. COuld you take a look?

Reply:
changed logic more scala way now is covered